pkg/cloud/vsphere/services/userdata/controlplane.go

```diff
@@ -24,6 +24,28 @@ import (
 )
 
 const (
+	cloudConfig = `[Global]
+insecure-flag = "1" # set to 1 if the vCenter uses a self-signed cert
+datacenters = "{{ .Datacenter }}"
+
+[VirtualCenter "{{ .Server }}"]
+user = "{{ .User }}"
+password = "{{ .Password }}"
+
+[Workspace]
+server = "{{ .Server }}"
+datacenter = "{{ .Datacenter }}"
+folder = "{{ .ResourcePool }}"
```

Comment:
This actually needs to be a "VM Folder" - not the same as the Resource Pool: https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/existing.html#supported-parameters

So for example, one might place running VMs in the RP `cluster01/Resources` but the "VM Folder" they live in is purely organisation and could be something like `Testing/K8s/cluster-name`.

Reply:
The cloud config here was taken from what was set on CAPV previously https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/0ec7a864653df44316d6747601d8d2927cb0f3ea/pkg/cloud/vsphere/provisioner/common/templates.go#L311-L331 which has the resource pool as the folder, I'm just going to assume that was also a bug, will open a bug shortly, thanks Myles!